### PR TITLE
Add at::_safe_softmax op

### PR DESCRIPTION
**Pull request by jianyizh**
fuse safe softmax https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/transformers/attention.cpp#L656

### CODE REVIEW

File: test/xpu/nn/test_safe_softmax.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import torch
+from torch.testing._internal.common_utils import TestCase
+
+cpu_device = torch.device("cpu")
+xpu_device = torch.device("xpu")
+
+
+class TestSafeSoftMax(TestCase):
```
Comment (xytintel) [inline test/xpu/nn/test_safe_softmax.py:8]: If it is not present in the stock PyTorch tests, please move it to `test/regression/`

Comment (jianyizh) [reply test/xpu/nn/test_safe_softmax.py]: done

**Review by xytintel — APPROVED** [review]
